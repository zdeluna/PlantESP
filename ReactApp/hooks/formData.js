import {useState, useCallback} from 'react';

const formData = values => {
    const [formValues, setFormValues] = useState({
        ...values,
    });

    const handleFormValueChange = (key, value) => {
        setFormValues({
            ...formValues,
            [key]: value,
        });
    };

    return {formValues, handleFormValueChange};
};

export default formData;
