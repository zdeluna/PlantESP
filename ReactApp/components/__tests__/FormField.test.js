import React from 'react';
import renderer from 'react-test-renderer';
import FormField from '../../components/FormField';
import formData from '../../hooks/formData';
import {TextInput} from 'react-native';
import {renderHook, act} from '@testing-library/react-hooks';

describe('FormField', () => {
    const testFormObject = {email: 'test@gmail.com'};

    test('Text input is rendered', () => {
        const {result} = renderHook(() => formData(testFormObject));

        const testRenderer = renderer.create(
            <FormField
                label="Email"
                formKey="email"
                placeholder="Your email."
                handleFormValueChange={result.current.handleFormValueCahnge}
            />,
        );

        expect(testRenderer.toJSON()).toMatchSnapshot();
    });
});
