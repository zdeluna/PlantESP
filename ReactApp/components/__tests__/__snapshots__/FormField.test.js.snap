// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField Text input is rendered 1`] = `
<View
  style={Object {}}
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "marginBottom": 12,
        "paddingLeft": 10,
        "paddingTop": 10,
      }
    }
  >
    Email
  </Text>
  <TextInput
    allowFontScaling={true}
    onChange={[Function]}
    placeholder="Your email."
    rejectResponderTermination={true}
    style={
      Object {
        "borderRadius": 15,
        "borderWidth": 1,
        "fontSize": 20,
        "padding": 12,
      }
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
